File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = 0
        mx = nums[0]
        for num in nums[1:]:
            res = max(res, (mx - 1) * (num - 1))
            mx = max(mx, num)
        return res

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int res = 0;
        int mx = nums[0];
        for(int i = 1; i < nums.size(); i++)
        {
            res = max(res, (mx - 1) * (nums[i] - 1));
            mx = max(mx, nums[i]);
        }
        return res;
    }
};